test/tokenbridge.test.ts

```diff
@@ -0,0 +1,204 @@
+ * This is done to strip leading zeros.
+ * @param address a hex string representation of an address
+ * @returns an adapted hex string representation of the address
+ */
+function adaptAddress(address: string) {
+  return "0x" + BigInt(address).toString(16);
+}
+
+/**
+ * Expects address equality after adapting them.
+ * @param actual 
+ * @param expected 
+ */
+function expectAddressEquality(actual: string, expected: string) {
+  expect(adaptAddress(actual)).to.equal(adaptAddress(expected));
+}
+
+/**
+ * Follows the example at https://www.cairo-lang.org/docs/hello_starknet/l1l2.html
+ * Shows the communication between an L2 contract defined in l1l2.cairo
+ * and an L1 contract defined in https://www.cairo-lang.org/docs/_static/L1L2Example.sol
+ */
+describe('TokenBridge', async function() {
+  this.timeout(TIMEOUT);
+ 
+  let l1user: SignerWithAddress;
+  let l2user: Account;
+  let signer: SignerWithAddress;
+  const networkUrl: string = (network.config as HttpNetworkConfig).url;
+  console.log(networkUrl)
+  // L2
+  let L2TokenFactory: StarknetContractFactory;
+  let l2tokenA: StarknetContract;
+  let l2tokenB: StarknetContract;
+  let TokenBridgeL2: StarknetContractFactory;
+  let tokenBridgeL2: StarknetContract;
+  // L1
+  let MockStarknetMessaging: ContractFactory;
+  let mockStarknetMessaging: Contract;
+  let L1TokenFactory: ContractFactory;
+  let l1tokenA: Contract;
+  let l1tokenB: Contract;
+  let TokenBridgeL1: ContractFactory;
+  let tokenBridgeL1: Contract;
+  let messagingContractAddress: string;
+
+  before(async function () {
+
+    // L2 deployments
+
+    l2user = await starknet.deployAccount("OpenZeppelin");
+
+    L2TokenFactory = await starknet.getContractFactory('l2_token');
+    l2tokenA = await L2TokenFactory.deploy(
+        { name: 1234, symbol: 123, decimals: 18, minter_address: BigInt(l2user.starknetContract.address) });
+    l2tokenB = await L2TokenFactory.deploy(
+        { name: 4321, symbol: 321, decimals: 18, minter_address: BigInt(l2user.starknetContract.address) });
+  
+    TokenBridgeL2 = await starknet.getContractFactory('token_bridge');
+    tokenBridgeL2 = await TokenBridgeL2.deploy({ governor_address: BigInt(l2user.starknetContract.address) });
+
+    // L1 deployments
+
+    [signer, l1user] = await ethers.getSigners();
+
+    MockStarknetMessaging = await ethers.getContractFactory(
+      'MockStarknetMessaging',
+      signer,
+    );
+    mockStarknetMessaging = await MockStarknetMessaging.deploy();
+    await mockStarknetMessaging.deployed();
+
+    L1TokenFactory = await ethers.getContractFactory('L1Token', signer);
+    l1tokenA = await L1TokenFactory.deploy(1000);
+    l1tokenB = await L1TokenFactory.deploy(1000);
+
+    TokenBridgeL1 = await ethers.getContractFactory('TokenBridge', signer);
+    tokenBridgeL1 = await TokenBridgeL1.deploy();
+    await tokenBridgeL1.deployed();
+  });
+
+  it('should deploy the messaging contract', async () => {
```

Comment:
We don't really need this test, it was used by hardhat-plugin to prove to itself that it was working. We should just deploy the messaging contract in a before